superset-frontend/src/SqlLab/components/TemplateParamsEditor.jsx

```diff
@@ -68,23 +60,27 @@ export default class TemplateParamsEditor extends React.Component {
     } catch (e) {
       isValid = false;
     }
-    this.setState({ parsedJSON, isValid, codeText });
+    setState({ parsedJSON, isValid, codeText });
     const newValue = isValid ? codeText : '{}';
-    if (newValue !== this.props.code) {
-      this.props.onChange(newValue);
+    if (newValue !== code) {
+      onChange(newValue);
     }
-  }
+  };
+
+  useEffect(() => {
+    onChange(codeText);
+  }, [codeText]);
 
-  renderDoc() {
-    return (
+  const modalBody = (
+    <div>
       <p>
-        {t('Assign a set of parameters as')}
+        'Assign a set of parameters as'
```

Comment:
These probably shouldn't have quotes here anymore. Did you validate the change to make sure the test renders properly?

Reply:
I removed the single quotes in the most recent commit, it works properly locally. I ran the tests in SqlEditor_spec:
<img width="376" alt="Screen Shot 2021-02-01 at 7 17 31 PM" src="https://user-images.githubusercontent.com/55605634/106539640-fa8ee080-64c3-11eb-8556-f0b786d03516.png">

but I didn't see any tests specifically for TemplateParamsEditor.

Reply:
lol, this was a typo, i meant to say "make sure the _text_ renders properly"